core/src/main/java/org/elasticsearch/index/analysis/CustomAnalyzerProvider.java

```
@@ -77,13 +78,20 @@ public void build(AnalysisService analysisService) {
             tokenFilters.add(tokenFilter);
         }
 
-        int positionOffsetGap = analyzerSettings.getAsInt("position_offset_gap", 0);
-        int offsetGap = analyzerSettings.getAsInt("offset_gap", -1);
+        int positionIncrementGap = 0;
+
+        if (Version.indexCreated(indexSettings).before(Version.V_2_0_0) ){
+            positionIncrementGap = analyzerSettings.getAsInt("position_offset_gap", 0);
+        }else{
```

I would throw an exception if both are specified.